[DOC] Fix style issues and add videos to tracing and profiling shared files (#81215)
grafana/grafana: docs/sources/datasources/grafana-pyroscope.md

# Grafana Pyroscope data source

> Formerly Phlare, now Grafana Pyroscope, a horizontally scalable, highly available, multi-tenant, OSS, continuous profiling aggregation system. Add it as a data source, and you are ready to query your profiles in [Explore][explore].

Reviewer: ```suggestion
Grafana Pyroscope is a horizontally scalable, highly available, multi-tenant, OSS, continuous profiling aggregation system. Add it as a data source, and you are ready to query your profiles in [Explore][explore].
```

   | -------------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
   | `Name`         | A name to specify the data source in panels, queries, and Explore.                                                                                                                                                                                                                                                        |
   | `Default`      | The default data source will be pre-selected for new panels.                                                                                                                                                                                                                                                              |

>    | `URL`          | The URL of the Grafana Pyroscope instance, for example, `http://localhost:4100`                                                                                                                                                                                                                                        |

Reviewer 2: ```suggestion
   | `URL`          | The URL of the Grafana Pyroscope instance, for example, `http://localhost:4100`.                                                                                                                                                                                                                                       |
```

Or does it add confusion if there's a full-stop here?

Author: Nope. The other rows in the tables have periods, so I think that is good

>    | `URL`          | The URL of the Grafana Pyroscope or Phlare instance, e.g., `http://localhost:4100`                                                                                                                                                                                                                                        |

Reviewer: ```suggestion
   | `URL`          | The URL of the Grafana Pyroscope instance, e.g., `http://localhost:4100`                                                                                                                                                                                                                                        |
```

To access the query editor:

1.  Sign into Grafana or Grafana Cloud.
1.  Select your Pyroscope data source.
1.  From the menu, choose **Explore**.

1.  Select a profile type from the drop-down menu.

    {{< figure src="/media/docs/pyroscope/query-editor/select-profile.png" class="docs-image--no-shadow" max-width="450px" caption="Profile selector" >}}

1.  Use the labels selector input to filter by labels. Pyroscope uses similar syntax to Prometheus to filter labels.
    Refer to [Pyroscope documentation](https://grafana.com/docs/pyroscope/latest/) for available operators and syntax.

        While the label selector can be left empty to query all profiles without filtering by labels, the profile type or app must be selected for the query to be valid.

        Grafana doesn't show any data if the profile type or app isn’t selected when a query runs.

    ![Labels selector](/media/docs/pyroscope/query-editor/labels-selector.png 'Labels selector')

1.  Expand the **Options** section to view **Query Type** and **Group by**.
    ![Options section](/media/docs/pyroscope/query-editor/options-section.png 'Options section')

> 1.  Select a query type to return the profile data which can be shown in the [Flame Graph][flame-graph], metric data visualized in a graph, or both. You can only select both options in a dashboard, because panels allow only one visualization.

Reviewer: is there supposed to be a link here?

Reviewer 3: This is a less common version of Markdown link known as [_reference-style_](https://spec.commonmark.org/0.30/#link-reference-definitions).

The destination is defined in https://github.com/grafana/grafana/pull/81215/files/4eb82167a07ec87480c6330fd32801203de56679#diff-74748b5f590531c97ca006dcf25af41006921b78039b7fe41754e62f3cd66a3eR129-R130.

   | `Basic Auth`   | Enable basic authentication to the data source.                                                                                                                                                                                                                                                                          |
   | `User`         | User name for basic authentication.                                                                                                                                                                                                                                                                                      |
   | `Password`     | Password for basic authentication.                                                                                                                                                                                                                                                                                       |

>    | `Minimal step` | Used for queries returning timeseries data. Phlare backend, similar to Prometheus, scrapes profiles at certain intervals. To prevent querying at smaller interval, use Minimal step same or higher than your Phlare scrape interval. For Pyroscope backend, this prevents returning too many data points to the frontend. |

Author: @Rperry2174 Do we need to do anything with the two sentences that refer to Phlare on line 45? Should they be updated in ay way or removed?

Reviewer: yeah we should replace phlare with pyroscope everywhere here (and remove any remaining references to phlare

Author: I searched through the Pyroscope page and didn't find any more references to Phlare.

To access the query editor:

1.  Sign into Grafana or Grafana Cloud.
1.  Select your Pyroscope data source.
1.  From the menu, choose **Explore**.

1.  Select a profile type from the drop-down menu.

    {{< figure src="/media/docs/pyroscope/query-editor/select-profile.png" class="docs-image--no-shadow" max-width="450px" caption="Profile selector" >}}

1.  Use the labels selector input to filter by labels. Pyroscope uses similar syntax to Prometheus to filter labels.
    Refer to [Pyroscope documentation](https://grafana.com/docs/pyroscope/latest/) for available operators and syntax.

        While the label selector can be left empty to query all profiles without filtering by labels, the profile type or app must be selected for the query to be valid.

>         Grafana doesn't show any data if the profile type or app isn’t selected when a query runs.

Reviewer 3: With this level of indentation, this section is rendered as a code block.

Author: it's weird because it looked okay in the preview and didn't appear as a code block. I should try it again and verify. 
Happy to fix it.

Reviewer 3: <img width="1624" alt="Screenshot 2024-01-25 at 21 24 11" src="https://github.com/grafana/grafana/assets/4599384/19d04b72-17d0-4f6b-86f1-0947e3ee6a20">

Probably changed as a result of running `prettier` when the file was in an ambiguous form. 

I'd expect the indentation of the paragraphs to match the indentation of the first character of the list item.

For example:

```markdown
1. List item.

   Paragraph as part of the first list item.
```

Author: I think you're right that it was Prettier that added the spaces. They should be fixed now. I removed the extra spaces and put things back in line.

    ![Labels selector](/media/docs/pyroscope/query-editor/labels-selector.png 'Labels selector')

1.  Expand the **Options** section to view **Query Type** and **Group by**.
    ![Options section](/media/docs/pyroscope/query-editor/options-section.png 'Options section')

> 1.  Select a query type to return the profile data which can be shown in the [Flame Graph][flame-graph], metric data visualized in a graph, or both. You can only select both options in a dashboard, because panels allow only one visualization.

Reviewer 3: The spacing in this section looks off.

There are two spaces after each list item (which seems to render fine but is unusual).

Author: I'm not sure why. I'll check the formatting and fix it. Thank you.

Author: It should be good now.
